### ISSUE TITLE
No device found 

### ISSUE THREAD

**Issue by maxmib** [custom-components/ble_monitor#327]
Home Assistant 2021.3.4  python3.9  venv 
After setting ,no device found .But my machine can find lots of devices when I use command "hcitool lescan " manually.
 Debug log in homeassistant is below .
 

2021-04-17 02:33:23 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_init user_input: None
2021-04-17 02:33:23 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_init (before): {'bt_interface': ['43:45:C0:00:1F:AC'], 'period': 10, 'discovery': False, 'active_scan': False, 'report_unknown': False, 'batt_entities': True, 'decimals': 1, 'log_spikes': True, 'use_median': False, 'restore_state': False, 'devices': [{'mac': '50:EC:50:CE:6A:E4', 'encryption_key': 'xxxxxxxxxxxxxxxxxxxxxxxxxx', 'temperature_unit': '°C', 'decimals': 'default', 'use_median': 'default', 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'E7:2E:00:92:B0:84', 'temperature_unit': '°C', 'decimals': 'default', 'use_median': 'default', 'restore_state': 'default', 'reset_timer': 0}], 'is_flow': True, 'hci_interface': [0]}
2021-04-17 02:33:31 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_init user_input: None
2021-04-17 02:33:31 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_init (before): {'bt_interface': ['43:45:C0:00:1F:AC'], 'period': 10, 'discovery': False, 'active_scan': False, 'report_unknown': False, 'batt_entities': True, 'decimals': 1, 'log_spikes': True, 'use_median': False, 'restore_state': False, 'devices': [{'mac': '50:EC:50:CE:6A:E4', 'encryption_key': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'temperature_unit': '°C', 'decimals': 'default', 'use_median': 'default', 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'E7:2E:00:92:B0:84', 'temperature_unit': '°C', 'decimals': 'default', 'use_median': 'default', 'restore_state': 'default', 'reset_timer': 0}], 'is_flow': True, 'hci_interface': [0]}
2021-04-17 02:33:31 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor] 1 encryptors mac:key pairs loaded.
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: [50:EC:50:CE:6A:E4, E7:2E:00:92:B0:84]
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor] 2 whitelist item(s) loaded.
2021-04-17 02:33:32 DEBUG (Thread-2895) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-17 02:33:32 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-17 02:33:41 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor] 1 encryptors mac:key pairs loaded.
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: [50:EC:50:CE:6A:E4, E7:2E:00:92:B0:84]
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor] 2 whitelist item(s) loaded.
2021-04-17 02:33:42 DEBUG (Thread-2901) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-17 02:33:42 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-17 02:33:51 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor] 1 encryptors mac:key pairs loaded.
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: [50:EC:50:CE:6A:E4, E7:2E:00:92:B0:84]
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor] 2 whitelist item(s) loaded.
2021-04-17 02:33:52 DEBUG (Thread-2907) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-17 02:33:52 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).

**Comment by Ernst79** [comment]
When you use `hcitool lescan`, can it find data of these two MAC addresses? BLE monitor logs say that it didn't receive any data which follows Xiaomi MiBeacon, Qingping or ATC format. "hcitool lescan" will find many more devices, like phones, PC's etc. 

Try to find the MAC address (reversed per two characters) of the sensors by looking in the hcitool lescan data. So, for 50:EC:50:CE:6A:E4, it will be reported as E4 6A CE 50 EC 50. Check if you can find data, if not, I suspect reception issues (either sensor too far away, or an SSD drive blocking the signal normally)

**Comment by maxmib** [comment]
> When you use `hcitool lescan`, can it find data of these two MAC addresses? BLE monitor logs say that it didn't receive any data which follows Xiaomi MiBeacon, Qingping or ATC format. "hcitool lescan" will find many more devices, like phones, PC's etc. 
> 
> Try to find the MAC address (reversed per two characters) of the sensors by looking in the hcitool lescan data. So, for 50:EC:50:CE:6A:E4, it will be reported as E4 6A CE 50 EC 50. Check if you can find data, if not, I suspect reception issues (either sensor too far away, or an SSD drive blocking the signal normally)

Data of these two mac address can be found in lescan data.And compared with others' debug logs ,there is not 'hcidump:connected to hci0 'in mine.Can this be the problem?

**Comment by Ernst79** [comment]
Yes, that can be a problem. The option `hci_interface` will overrule `bt_interface`, when used both at the same time.

**Comment by maxmib** [comment]
> Yes, that can be a problem. The option `hci_interface` will overrule `bt_interface`, when used both at the same time. 

I only use web interface to configure this integration.So the option bt_interface works.But still no idea about why no device can be found.

**Comment by Ernst79** [comment]
Ah, sorry. I thought you use YAML configuration. In case of configuration in the UI, it is ok to see both `hci_interface` and `bt_interface`. But they should refer to the same device. Can you show the output of the following command: `hcitool dev`. It normally shows something like this 

```
Devices:
        hci0    B8:27:EB:77:75:50
```

And did you follow [step 1 of the installation process](https://github.com/custom-components/ble_monitor/#how-to-install)



this step is especially important for venv installations. Also check this [FAQ article](https://github.com/custom-components/ble_monitor/blob/master/faq.md#i-get-a-permissionerror-in-home-assistant-after-the-installation-or-python-upgrade), which might help. If the venv is created with a different python version than used by HA, this article might solve your problem.

**Comment by maxmib** [comment]
![1618671094(1)](https://user-images.githubusercontent.com/30944505/115117109-8e565000-9fcf-11eb-8589-b2d4e440ef61.jpg)

![1618671183(1)](https://user-images.githubusercontent.com/30944505/115117143-b47bf000-9fcf-11eb-9fa2-dbaca128622b.jpg)

output after command sudo getcap `readlink -f \`which python3.9\``
and the venv is created with  same python version than used by HA ,which is python 3.9
![1618671426(1)](https://user-images.githubusercontent.com/30944505/115117274-4683f880-9fd0-11eb-8e7b-ac08965c7e0c.jpg)

**Comment by Ernst79** [comment]
> Data of these two mac address can be found in lescan data.And compared with others' debug logs ,there is not 'hcidump:connected to hci0 'in mine. Can this be the problem?

This is definitely weird. It should normally show this message, and it doesn't show an error. 

Could you try to change the following code in `__init__.py` line 800-805

```
            for hci in self._interfaces:
                try:
                    mysocket[hci] = aiobs.create_bt_socket(hci)
                except OSError as error:
                    _LOGGER.error("HCIdump thread: OS error (hci%i): %s", hci, error)
                else:
```

to 

```
            for hci in self._interfaces:
                _LOGGER.debug("hci interface is (hci%i)", hci)
                try:
                    mysocket[hci] = aiobs.create_bt_socket(hci)
                except Exception as error:
                    _LOGGER.error("HCIdump thread: Error (hci%i): %s", hci, error)
                else:
```

**Comment by maxmib** [comment]
> > Data of these two mac address can be found in lescan data.And compared with others' debug logs ,there is not 'hcidump:connected to hci0 'in mine. Can this be the problem?
> 
> This is definitely weird. It should normally show this message, and it doesn't show an error.
> 
> Could you try to change the following code in `__init__.py` line 800-805
> 
> ```
>             for hci in self._interfaces:
>                 try:
>                     mysocket[hci] = aiobs.create_bt_socket(hci)
>                 except OSError as error:
>                     _LOGGER.error("HCIdump thread: OS error (hci%i): %s", hci, error)
>                 else:
> ```
> 
> to
> 
> ```
>             for hci in self._interfaces:
>                 _LOGGER.debug("hci interface is (hci%i)", hci)
>                 try:
>                     mysocket[hci] = aiobs.create_bt_socket(hci)
>                 except Exception as error:
>                     _LOGGER.error("HCIdump thread: Error (hci%i): %s", hci, error)
>                 else:
> ```

this is the debug log 

2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] HCI interface is [0]
2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] 1 encryptors mac:key pairs loaded.
2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: [50:EC:50:CE:6A:E4, E7:2E:00:92:B0:84]
2021-04-18 01:11:21 DEBUG (MainThread) [custom_components.ble_monitor] 2 whitelist item(s) loaded.
2021-04-18 01:11:21 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 01:11:21 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 01:11:21 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'
2021-04-18 01:11:21 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 01:11:25 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Starting binary sensor entry startup
2021-04-18 01:11:25 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialization
2021-04-18 01:11:25 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialized
2021-04-18 01:11:25 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary sensor entry setup finished
2021-04-18 01:11:25 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary entities updater loop started!
2021-04-18 01:11:33 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Starting measuring sensor entry startup
2021-04-18 01:11:33 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialization
2021-04-18 01:11:33 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialized
2021-04-18 01:11:33 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Measuring sensor entry setup finished
2021-04-18 01:11:33 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Entities updater loop started!
2021-04-18 01:11:38 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 01:11:43 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while stop scan request on hci0
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while closing connection on hci0
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 01:11:43 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'
2021-04-18 01:11:43 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 01:11:49 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 01:11:53 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while stop scan request on hci0
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while closing connection on hci0
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 01:11:53 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'
2021-04-18 01:11:53 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 01:11:59 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 01:12:04 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while stop scan request on hci0
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while closing connection on hci0
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 01:12:04 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'
2021-04-18 01:12:04 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 01:12:10 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 01:12:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while stop scan request on hci0
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Key error while closing connection on hci0
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 01:12:14 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'
2021-04-18 01:12:14 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop

**Comment by Ernst79** [comment]
Ah, you have the famous AF_BLUETOOTH error

`HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'`

The issue and solution is explained here in the [FAQ](https://github.com/custom-components/ble_monitor/blob/master/faq.md#i-get-attributeerror-module-socket-has-no-attribute-af_bluetooth-in-home-assistant-after-the-installation-or-python-upgrade). You are not the first. 

Basically, the error is that you have created a python installation without Bluetooth support. The FAQ will give you instructions how you should setup python to include bluetooth (socket support).

**Comment by maxmib** [comment]
> Ah, you have the famous AF_BLUETOOTH error
> 
> `HCIdump thread: Error (hci0): module 'socket' has no attribute 'AF_BLUETOOTH'`
> 
> The issue and solution is explained here in the [FAQ](https://github.com/custom-components/ble_monitor/blob/master/faq.md#i-get-attributeerror-module-socket-has-no-attribute-af_bluetooth-in-home-assistant-after-the-installation-or-python-upgrade). You are not the first.
> 
> Basically, the error is that you have created a python installation without Bluetooth support. The FAQ will give you instructions how you should setup python to include bluetooth (socket support).

this is the new debug log after rebuilding python

2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] HCI interface is [0]
2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] 1 encryptors mac:key pairs loaded.
2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: [50:EC:50:CE:6A:E4, E7:2E:00:92:B0:84]
2021-04-18 02:52:58 DEBUG (MainThread) [custom_components.ble_monitor] 2 whitelist item(s) loaded.
2021-04-18 02:52:58 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 02:52:58 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 02:52:58 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-04-18 02:53:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Starting binary sensor entry startup
2021-04-18 02:53:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialization
2021-04-18 02:53:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialized
2021-04-18 02:53:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary sensor entry setup finished
2021-04-18 02:53:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary entities updater loop started!
2021-04-18 02:53:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Starting measuring sensor entry startup
2021-04-18 02:53:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialization
2021-04-18 02:53:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialized
2021-04-18 02:53:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Measuring sensor entry setup finished
2021-04-18 02:53:14 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Entities updater loop started!
2021-04-18 02:53:24 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:53:24 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:53:24 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while sending scan request on hci0: Event loop stopped before Future completed.
2021-04-18 02:53:24 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 02:53:34 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:53:34 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:53:34 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 02:53:45 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:53:45 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:53:45 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while stop scan request on hci0: Event loop stopped before Future completed.
2021-04-18 02:53:45 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 02:53:45 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 02:53:45 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 02:53:45 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 02:53:45 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-04-18 02:53:55 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:53:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:53:55 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while sending scan request on hci0: Event loop stopped before Future completed.
2021-04-18 02:53:55 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 02:54:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:54:06 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:54:06 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 02:54:16 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:54:16 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:54:16 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while stop scan request on hci0: Event loop stopped before Future completed.
2021-04-18 02:54:16 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-04-18 02:54:16 DEBUG (Thread-4) [custom_components.ble_monitor] 0 HCI events processed for previous period.
2021-04-18 02:54:16 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-04-18 02:54:16 DEBUG (Thread-4) [custom_components.ble_monitor] hci interface is (hci0)
2021-04-18 02:54:16 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-04-18 02:54:26 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:54:26 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:54:26 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while sending scan request on hci0: Event loop stopped before Future completed.
2021-04-18 02:54:26 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-04-18 02:54:36 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:54:36 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:54:36 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-04-18 02:54:46 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-04-18 02:54:46 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 MiBeacon BLE ADV messages processed for 0 measuring device(s).
2021-04-18 02:54:46 ERROR (Thread-4) [custom_components.ble_monitor] HCIdump thread: Runtime error while stop scan request on hci0: Event loop stopped before Future completed.

**Comment by Ernst79** [comment]
Did you try this?

```
sudo setcap 'cap_net_raw,cap_net_admin+eip' `readlink -f \`which python3.9\``
```

**Comment by maxmib** [comment]
> Did you try this command, assuming HA is running in python 3.9
> 
> ```
> sudo setcap 'cap_net_raw,cap_net_admin+eip' `readlink -f \`which python3.9\``
> ```
After run setcap 'cap_net_raw,cap_net_admin+eip' `readlink -f \`which python3.9\`` under root user ,it works .
And I want to make sure that this command should run under root user or run after switching to venv ?

ps: the solution for   Warning Log: Cannot set filter: libpcap is not available. Cannot compile filter ! 
https://github.com/home-assistant/core/issues/45846

**Comment by Ernst79** [comment]
You run it as root. ThE command gives Python rootless access to the HCI interface
